connector_test.go

```diff
@@ -123,37 +117,104 @@ var volumeConfig = `
 }
 `
 
-func TestSimpleVolume(t *testing.T) {
-	logger := log.NewTestLogger(t)
-	fileContent, err := os.ReadFile("./tests/volume/test_file.txt")
-	assert.NoError(t, err)
-
-	connector, _ := getConnector(t, volumeConfig)
-	cwd, err := os.Getwd()
-	assert.NoError(t, err)
-	// disable selinux on the test folder in order to make the file readable from within the container
-	cmd := exec.Command("chcon", "-Rt", "svirt_sandbox_file_t", fmt.Sprintf("%s/tests/volume", cwd)) //nolint:gosec
-	err = cmd.Run()
-	if err != nil {
-		logger.Warningf("failed to set SELinux permissions on folder, chcon error: %s, this may cause test failure if SELinux is enabled.", err.Error())
-	}
-
-	container, err := connector.Deploy(
+// bindMountHelper is a helper function which tests plugins with a file
+// bind-mounted inside the container.  Options for the mount and the expected
+// outcome of the test are provided by parameters.  The test creates a
+// temporary file containing appropriate content, configures that file to be
+// mounted inside the container, and then starts the plugin; the test then
+// tells the plugin to output the contents of the mapped file and checks it
+// against the value originally written to the file.
+func bindMountHelper(t *testing.T, options string, expectedPass bool) {
+	fileContent := fmt.Sprintf("bind mount test with option %q\n", options)
+	mountFile := assert.NoErrorR[*os.File](t)(os.CreateTemp("", "bind_mount_test_*.txt"))
+	t.Cleanup(func() { assert.NoError(t, os.Remove(mountFile.Name())) })
+	assert.NoErrorR[int](t)(mountFile.WriteString(fileContent))
+	assert.NoError(t, mountFile.Close())
+	connector, _ := getConnector(t, fmt.Sprintf(volumeConfig, mountFile.Name(), options))
+
+	// Run the plugin
+	container := assert.NoErrorR[deployer.Plugin](t)(connector.Deploy(
 		context.Background(),
-		"quay.io/arcalot/podman-deployer-test-helper:0.1.0")
-	assert.NoError(t, err)
+		"quay.io/arcalot/podman-deployer-test-helper:0.1.0"))
+	t.Cleanup(func() { assert.NoError(t, container.Close()) })
+
+	// Tell the plugin to output the contents of the mapped file.
+	assert.NoErrorR[int](t)(container.Write([]byte("volume\n")))
 
-	var containerInput = []byte("volume\n")
-	_, err = container.Write(containerInput)
-	assert.NoError(t, err)
 	// Note: If it ends up with length zero buffer, restarting the VM may help:
 	// https://stackoverflow.com/questions/71977532/podman-mount-host-volume-return-error-statfs-no-such-file-or-directory-in-ma
-	readBuffer := readOutputUntil(t, container, string(fileContent))
-	assert.GreaterThan(t, len(readBuffer), 0)
+	readBuffer := readOutputUntil(t, container, fileContent)
+	if expectedPass {
+		assert.Contains(t, string(readBuffer), fileContent)
+	} else {
+		// If this assertion fails, then we actually found what we were looking
+		// for (despite our hopes to the contrary), so having the failure
```

Comment:
"found what we were looking for" is a bit misleading here. It's more like "what we would have been looking for had we expected success" (and, in this branch, definitely are *not* looking for) ... or just "the marker text we wrote to the file".

A comment that this is a weak test that doesn't attempt to verify *why* the "not expected" text is missing might also be in order, as long as you're writing an extended comment. 😆

Reply:
I've reworded the comment, and hopefully it is more forward-applicable, now.  (Much of the previous text should probably have been put in a review comment here instead of in the code.)  And, I added a note on the weakness of the test.